source/common/network/filter_manager_impl.cc

```diff
@@ -46,18 +46,19 @@ void FilterManagerImpl::onContinueReading(ActiveReadFilter* filter,
   }
 
   for (; entry != upstream_filters_.end(); entry++) {
-    if (!(*entry)->initialized_) {
+    if (!(*entry)->initialized_ && connection_.state() == Connection::State::Open) {
       (*entry)->initialized_ = true;
       FilterStatus status = (*entry)->filter_->onNewConnection();
-      if (status == FilterStatus::StopIteration) {
+      if (status == FilterStatus::StopIteration || connection_.state() != Connection::State::Open) {
         return;
       }
     }
 
     StreamBuffer read_buffer = buffer_source.getReadBuffer();
-    if (read_buffer.buffer.length() > 0 || read_buffer.end_stream) {
+    if ((read_buffer.buffer.length() > 0 || read_buffer.end_stream) &&
+        connection_.state() == Connection::State::Open) {
```

Comment:
Same here, shouldn't we only be checking after a filter runs if that is what we want to do?